server/api.go

```diff
@@ -208,6 +231,34 @@ func (a *apiServer) Publish(ctx context.Context, req *client.PublishRequest) (
 	return resp, err
 }
 
+func (a *apiServer) resumeStream(ctx context.Context, req *client.PublishRequest) error {
+	if req.Stream == "" {
+		return nil
+	}
+	stream := a.metadata.GetStream(req.Stream)
+	if stream == nil {
+		status.Error(codes.NotFound, fmt.Sprintf("No such stream: %s", req.Stream))
+	}
+	partition, ok := stream.partitions[0] // Only check the first partition for now
+	if !ok {
+		status.Error(codes.NotFound, fmt.Sprintf("No partition in stream: %s", req.Stream))
+	}
+	if partition.paused {
+		// This partition (and thus, this stream) is paused.
+		// We need to re-create the stream before being able to use it.
+		protoPart := partition.Partition
+		for i := 0; i < len(stream.partitions); i++ {
+			if e := a.metadata.CreatePartition(ctx, &proto.CreatePartitionOp{
+				Partition: protoPart,
```

Comment:
It looks like this is creating the same partition over and over since it's reusing the `protoPart` from `stream.partitions[0]`. This should use the correct partition:

```go
for _, partition := range stream.partitions {
        if e := a.metadata.CreatePartition(ctx, &proto.CreatePartitionOp{
                Partition: partition.Partition,
        }); e != nil {
                ...
        }
}
```

Reply:
Ha, yes, good catch. Should be fixed in 1bb7320.